config/management/src/secure_backend.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use libra_config::config::{OnDiskStorageConfig, SafetyRulesBackend, VaultConfig};
+use libra_secure_storage::Storage;
+use std::{collections::HashMap, convert::Into, path::PathBuf};
+
+/// SecureBackend is a parameter that is stored as a colon separated key/value pairs. The only
+/// expected key is backend which defines which of the SecureBackends the parameters refer to.
+/// Some backends require parameters others do not, so that requires a conversion into the
+/// SafetyRulesBackend type to parse.
+struct SecureBackend {
+    backend: String,
+    parameters: HashMap<String, String>,
+}
+
+impl SecureBackend {
+    const BACKEND: &'static str = "backend";
+}
+
+impl From<&str> for SecureBackend {
+    fn from(s: &str) -> Self {
+        let mut parameters = s
+            .split(';')
+            .map(|pair| {
+                let kv = pair.split('=').collect::<Vec<_>>();
+                if kv.len() != 2 {
+                    panic!("Invalid kv found in backend: {:?}", kv);
+                }
+                (kv[0].into(), kv[1].into())
+            })
+            .collect::<HashMap<String, String>>();
+        let backend = parameters.remove(Self::BACKEND).expect("Missing backend");
+        Self {
+            backend,
+            parameters,
+        }
+    }
+}
+
+impl Into<SafetyRulesBackend> for SecureBackend {
+    fn into(mut self) -> SafetyRulesBackend {
+        let backend = match self.backend.as_ref() {
+            "disk" => {
+                let mut config = OnDiskStorageConfig::default();
+                config.set_data_dir(PathBuf::from(""));
+                let path = self.parameters.remove("path");
+                let path = path.expect("Unable to retrieve path from disk backend");
+                config.path = PathBuf::from(path);
+                SafetyRulesBackend::OnDiskStorage(config)
+            }
+            "memory" => SafetyRulesBackend::InMemoryStorage,
+            "vault" => {
+                let server = self.parameters.remove("server");
```

Comment:
I'd also make these special strings constants, so it's super obvious what strings we're matching on.

Reply:
agreed, let's save that for a refactor?

Reply:
Sounds good!